lib/assert.js

```diff
@@ -608,6 +623,224 @@ assert.notStrictEqual = function notStrictEqual(actual, expected, message) {
+
+    for (let i = 0; i < safeIteratorArray.length; i++) {
+      if (!SetPrototypeHas(expected, safeIteratorArray[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  // Check if expected array is a subset of actual array
+  if (ArrayIsArray(actual) && ArrayIsArray(expected)) {
+    return assertIncludes(actual, expected, loose);
+  }
+
+  // Check non object types equality
+  if (!isPlainObject(actual) || !isPlainObject(expected)) {
+    if (isDeepEqual === undefined) {
+      lazyLoadComparison();
+    }
+    return loose ? isDeepEqual(actual, expected) : isDeepStrictEqual(actual, expected);
+  }
+
+  // Check if actual and expected are null or not objects
+  if (actual == null || expected == null) {
+    return false;
+  }
+
+  // Use Reflect.ownKeys() instead of Object.keys() to include symbol properties
+  const keysExpected = ReflectOwnKeys(expected);
+
+  // Handle circular references
+  if (SetPrototypeHas(comparedObjects, actual)) {
+    return true;
+  }
+  comparedObjects.add(actual);
+
+  // Check if all expected keys and values match
+  for (let i = 0; i < keysExpected.length; i++) {
+    const key = keysExpected[i];
+    assert(
+      ReflectHas(actual, key),
+      new AssertionError({ message: `Expected key ${String(key)} not found in actual object` }),
+    );
+    if (!compareBranch(actual[key], expected[key], loose, comparedObjects)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ * The strict equivalence assertion test between two objects
+ * @param {any} actual
+ * @param {any} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.deepMatchStrict = function deepMatchStrict(
+  actual,
+  expected,
+  message,
+) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (!compareBranch(actual, expected)) {
+    innerFail({
+      actual,
+      expected,
+      message,
+      operator: 'deepMatchStrict',
+      stackStartFn: deepMatchStrict,
+    });
+  }
+};
+
+/**
+ * The equivalence assertion test between two objects
+ * @param {any} actual
+ * @param {any} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.deepMatch = function deepMatch(actual, expected, message) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (!compareBranch(actual, expected, true)) {
+    innerFail({
+      actual,
+      expected,
+      message,
+      operator: 'deepMatch',
+      stackStartFn: deepMatch,
+    });
+  }
+};
+
+/**
+ * The inclusion assertion test between two arrays or strings
+ * @param {Array | string} actual
+ * @param {Array | string} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.includes = function includes(actual, expected, message) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (!assertIncludes(actual, expected, true)) {
```

Comment:
I suspect this function signature is such that the types of `actual` and `expected` need validation here, otherwise this call to `assertIncludes()` is at risk of generating TypeErrors.